src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/models/State.kt

```diff
@@ -0,0 +1,93 @@
+import org.elasticsearch.common.xcontent.XContentParser.Token
+import org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+
+data class State(
+    val id: String = NO_ID,
+    val version: Long = NO_VERSION,
+    val stateName: String,
+    val actions: List<Map<String, Any>>, // TODO: Implement List<Action>
+    val transitions: List<Map<String, Any>> // TODO: Implement List<Transition>
+) : ToXContent {
+
+    fun toXContent(builder: XContentBuilder): XContentBuilder =
+        toXContent(builder, ToXContent.EMPTY_PARAMS)
+
+    override fun toXContent(builder: XContentBuilder, params: ToXContent.Params): XContentBuilder {
+        builder
+            .startObject()
+                .field(NAME_FIELD, stateName)
+                .field(ACTIONS_FIELD, actions.toTypedArray())
+                .field(TRANSITIONS_FIELD, transitions.toTypedArray())
+            .endObject()
+        return builder
+    }
+
+    companion object {
+        const val NO_ID = ""
+        const val NO_VERSION = Versions.NOT_FOUND
+        const val NAME_FIELD = "state"
+        const val ACTIONS_FIELD = "actions"
+        const val TRANSITIONS_FIELD = "transitions"
+
+        @JvmStatic
+        @JvmOverloads
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser, id: String = NO_ID, version: Long = NO_VERSION): State {
+            lateinit var name: String
+            val actions: MutableList<Map<String, Any>> = mutableListOf() // TODO: Implement List<Action>
+            val transitions: MutableList<Map<String, Any>> = mutableListOf() // TODO: Implement List<Transition>
+
+            ensureExpectedToken(Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+
+                when (fieldName) {
+                    NAME_FIELD -> name = xcp.text()
+                    ACTIONS_FIELD -> {
+                        ensureExpectedToken(Token.START_ARRAY, xcp.currentToken(), xcp::getTokenLocation)
+                        while (xcp.nextToken() != Token.END_ARRAY) {
+                            // actions.add(Action.parse(xcp)) // TODO: Implement Action.parse()
+                        }
+                    }
+                    TRANSITIONS_FIELD -> {
+                        ensureExpectedToken(Token.START_ARRAY, xcp.currentToken(), xcp::getTokenLocation)
+                        while (xcp.nextToken() != Token.END_ARRAY) {
+                            // transitions.add(Transition.parse(xcp)) // TODO: Implement Transition.parse()
+                        }
+                    }
+                }
+            }
+
+            return State(
+                id,
+                version,
+                stateName = requireNotNull(name) { "State name is null" },
+                actions = requireNotNull(actions) { "State actions are null" },
```

Comment:
Remove requireNotNull, you are initializing actions as a mutable list, should never be null. And change to to actions.toList()

Reply:
Done.